pkg/cmd/cli/cmd/edit.go

```diff
@@ -211,7 +211,11 @@ func RunEdit(fullName string, f *clientcmd.Factory, out io.Writer, cmd *cobra.Co
 			if err != nil {
 				return err
 			}
-			updated, err := resource.NewHelper(info.Client, info.Mapping).Update(info.Namespace, info.Name, false, data)
+			namespace, err := info.ResourceMapping().Namespace(obj)
```

Comment:
Enforce it right after InfoForData - but remember, some objects don't have namespaces

> On May 21, 2015, at 5:31 PM, Fabiano Franz <notifications@github.com> wrote:
> 
> In pkg/cmd/cli/cmd/edit.go:
> 
> > @@ -211,7 +211,11 @@ func RunEdit(fullName string, f *clientcmd.Factory, out io.Writer, cmd *cobra.Co
> >  			if err != nil {
> >  				return err
> >  			}
> > -			updated, err := resource.NewHelper(info.Client, info.Mapping).Update(info.Namespace, info.Name, false, data)
> > +			namespace, err := info.ResourceMapping().Namespace(obj)
> If you mean cmdNamespace it's available here, but not in applyPatch and AddError (unless I pass it).
> 
> —
> Reply to this email directly or view it on GitHub.
>